src/reference.ts

```diff
@@ -1,6 +1,9 @@
+import { REFERENCE_CHILD_OF, REFERENCE_FOLLOWS_FROM } from './constants';
 import Span from './span';
 import SpanContext from './span_context';
 
+export type ReferenceType = typeof REFERENCE_CHILD_OF | typeof REFERENCE_FOLLOWS_FROM;
```

Comment:
I would just use strings, similar to how Format is handled.

In other languages we sometimes were able to define a type that was externally extensible, e.g. in Go. But in Java it's just a string, even though the [Format](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/propagation/Format.java#L36) is strongly typed.